server/lib/s3backup/__init__.py

```diff
@@ -0,0 +1,129 @@
+"""
+This module provides convenience functions that interface to lower-level services, provided by the boto3 module.
+"""
+import boto3
+import os
+import sys
+import glob
+import base64
+import hashlib
+import shutil
+import time
+import configtools
+from datetime import datetime
+from configparser import ConfigParser
+
+
+class Entry(object):
+    """
+    An Entry object consists of a name (of an S3 object) and the MD5 value of that object. It is used to create an object with the name and MD5 value, so that it can be compared with other similar objects.
+    """
+    def __init__(self, name, md5):
+        self.name = name
+        self.md5 = md5
+
+    def __eq__(self, other):
+        return self.name == other.name and self.md5 == other.md5
+
+
+class S3Connector(object):
+    def __init__(self, access_key_id, secret_access_key, endpoint_url, bucket_name):
+        self.s3connector = boto3.client('s3',
+                                        aws_access_key_id='{}'.format(
+                                            access_key_id),
+                                        aws_secret_access_key='{}'.format(
+                                            secret_access_key),
+                                        endpoint_url='{}'.format(endpoint_url))
+
+    def list_objects(self, Bucket):
+        return self.s3connector.list_objects_v2(Bucket=Bucket)
+
+    def put_object(self, Bucket, Key, Body, ContentMD5):
+        return self.s3connector.put_object(Bucket=Bucket, Key=Key,
+                Body=Body, ContentMD5=ContentMD5)
+
+    def head_bucket(self, Bucket):
+        return self.s3connector.head_bucket(Bucket=Bucket)
+
+    def get_object(self, Bucket, Key):
+        return self.s3connector.get_object(Bucket=Bucket, Key=Key)
+
+
+class MockS3Connector(object):
+    """
+    The mock object is used for unit testing. It provides a "connector" to the backend service that is implemented using the local filesystem, rather than dispatching to the real S3 backend service.
+    """
+
+    def __init__(self, access_key_id, secret_access_key, endpoint_url, bucket_name):
+        self.path = endpoint_url
+        self.bucket_name = bucket_name
+
+    def list_objects(self, Bucket):
+        ob_dict = {}
+        result_list = glob.glob(os.path.join(
+            self.path, Bucket, "*.tar.xz"))
```

Comment:
This is missing a controller piece: it should be "*/*.tar.xz"  I think.
DAMHIKT.